### ISSUE TITLE
Fork notification

### ISSUE THREAD

**Issue by hadess** [zeniko/unarr#7]
As #6 hasn't been looked at yet, and so that others can benefit from our downstream changes, you can find here changes related to evince and the RAR decompressor:
https://git.gnome.org/browse/evince/log/cut-n-paste/unarr

So far, in addition to the compilation fixes in the PR, I've also updated the LZMA SDK to the latest upstream version.

**Comment by selmf** [comment]
I'd be interested in using this but it probably breaks the zip and 7z support unarr provides. I also noticed that you didn't update the Ppmd8 code in the lzmasdk folder. It's not part of the LZMA SDK, but a more recent version is distributed with the current 7z source code.

**Comment by hadess** [comment]
We're only interested in using the RAR support, as the zip and 7z support is provided by libarchive for us.

But the fork is more of an on-going concern. If there is an upstream here, and patches are merged, I'd be happy to do the work to keep it up to date. In the meanwhile, I'll just update our fork.

**Comment by selmf** [comment]
I'm not sure if Zeniko is still around as an active developer as he has pretty much gone silent for the last two years, but [SumatraPDF](https://github.com/sumatrapdfreader/sumatrapdf) (for which unarr was originally written) is still very active and might be interested in receiving your patches.

**Comment by hadess** [comment]
> but SumatraPDF (for which unarr was originally written) is still very active and might be interested in receiving your patches.

They're not:
https://github.com/sumatrapdfreader/sumatrapdf/pull/751

**Comment by selmf** [comment]
Well, I would be interested. My unarr fork can be found here:

https://github.com/selmf/unarr

Feel free to give me a PR. ;)

Since you do an embedded build of a cherry-picked part of the whole project in evince we will need to talk about a few things like compiler flags and compatibility with the rest of the project, but that shouldn't be a deal breaker.

**Comment by zeniko** [comment]
Yeah, sorry for the long silence. As you've noted, I no longer have the time to continue working on unarr.

Give the time, I'd have loved to add at least RARv5 support (fixing issue #5), but for that or any other changes, I'd have to re-setup my testing infrastructure (over multiple platforms, with an appropriately large testing corpus, including expensive fuzzing runs) in order to keep confidence in the code.

So if any of you want to take over the project, feel free to do so. Else I'll just leave it in its current, well tested and mostly working form (it compiled with all the compilers I had available at the time).

**Comment by selmf** [comment]
Glad to see you're still around Zeniko.

Unarr still is working fine. I've been shipping it as decompression library for [www.yacreader.com](url) on Linux for two years now and the next release will also use it on MacOS and Windows.

Hadess' compile troubles are the result of a stricter set of compiler flags than the default unarr flags, probably in combination with the -Werror flag that promotes every warning to an error. Fixing them still is a good idea, but they should be fixed in all parts of the code, not just those relevant for rar extraction.

I don't mind continuing your work on this project - I have to maintain the code anyways. Still, keeping up with the quality of your code will be a challenge, so any information or advice you can give me on your testing process and corpus would be greatly appreciated.
